File: Cpp/1288_Remove_Covered_Intervals.cpp
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end(), [](vector<int>& a, vector<int>& b){
            if(a[0] == b[0]) return a[1] > b[1];
            return a[0] < b[0];
        });

        int result = 1;
        int index = 0;
        
        for(int i = 1; i < intervals.size(); ++i){
            if(intervals[i][1] <= intervals[index][1]) continue;
            ++result;
            index = i;
        }

        return result;
    }
};

File: Kotlin/1288_Remove_Covered_Intervals.kt
class Solution {
    fun removeCoveredIntervals(intervals: Array<IntArray>): Int {
        intervals.sortWith(compareBy<IntArray>{ it[0] }.thenByDescending{ it[1] })

        var result = 1
        var index = 0
            
        for(i in intervals.indices){
            if(intervals[i][1] <= intervals[index][1]) continue
            ++result
            index = i
        }
        
        return result
    }
}